app/models/parser/tiktok_item.rb

```diff
@@ -22,10 +23,21 @@ def parse_data_for_parser(doc, _original_url, _jsonld_array)
 
       handle_exceptions(StandardError) do
         @parsed_data[:raw][:api] = @parsed_data[:raw][:oembed] = get_tiktok_api_data(url)
-        match = url.match(TIKTOK_ITEM_URL)
+        
+        if url.match(TIKTOK_ITEM_URL)
+          match = url.match(TIKTOK_ITEM_URL)
+          username = match['username']
+          external_id = match['id']
+        end
```

Comment:
suggestion: I think an elsif might be a little clearer here, if we expect it to be one or the other